tests/accelerators/test_common.py

```diff
@@ -44,3 +59,80 @@ def test_evaluate(tmpdir, trainer_kwargs):
     # make sure weights didn't change
     new_weights = model.layer_0.weight.clone().detach().cpu()
     torch.testing.assert_allclose(old_weights, new_weights)
+
+
+def test_model_parallel_setup_called(tmpdir):
+
+    class TestModel(BoringModel):
+
+        def on_model_parallel_setup(self):
```

Comment:
Add a check that the context manager is actually being called.

Reply:
Thanks, it made me see an Issue in implementation. Fixinggg

Reply:
Might want to check the hook is not called within the callbacks.

Reply:
I think we can assume Users would be aware of the implications.